oauth2_http/java/com/google/auth/oauth2/S2A.java

```diff
@@ -81,14 +81,9 @@ public final class S2A {
     this.config = getS2AConfigFromMDS();
   }
 
-  /** @return the mTLS S2A Address from the mTLS config. */
-  public String getMtlsS2AAddress() {
-    return config.getMtlsAddress();
-  }
-
-  /** @return the plaintext S2A Address from the mTLS config. */
-  public String getPlaintextS2AAddress() {
-    return config.getPlaintextAddress();
+  /** @return the cached S2AConfig. */
+  public S2AConfig getConfigFromMDS() {
+    return config;
   }
```

Comment:
What I had in mind was
```
public SecureSessionAgentConfig getConfig() {
  return getS2AConfigFromMDS();
}
```
as this `getConfig()` call is the explicit call that will make the network call.

Reply:
I see, so right now the config is being set in the builder and is a member of the class. The current setup ensures that only one call to MDS is made (since there is no need to make the request to MDS multiple times) no matter how many times a user calls getConfigFromMds.

I'm fine with this though, it shifts responsibility to the user to store config and not call getConfig() more than once to save latency of their application. This seems aligned with what you mentioned in our offline discussion yesterday about how it should be explicit a network call is being made.

Done in: https://github.com/googleapis/google-auth-library-java/pull/1573/commits/381efe4531e481a2f4a36fbaa70dc94ce55ecc38